app/api/userData/route.js

```diff
@@ -41,6 +66,23 @@ export async function POST(req) {
             await setDoc(doc(db, "userdata", userData.uid), userData.newUserData);
             return NextResponse.json('success', { status: 200 })
         };
+        if (userData.action === "savePurchase") {
+            const purchaseRecords = userData.cart.map((prod) => {
+                const purchasedProduct = {
+                    userID: userData.uid,
+                    quantity: prod.quantity,
+                    date: userData.date,
+                    productID: prod.asin,
+                };
+                return purchasedProduct
+            })
+            await Promise.all(
+                purchaseRecords.map((purchaseRecord) => {
+                    return addDoc(collection(db, 'purchases'), purchaseRecord);
```

Comment:
Does someone add multiple purchases at the same time often? Is there no `addDocs` function in firebase? In relational databases, you can do bulk inserts where you can do all of your inserts in one singular operation.